lib/matplotlib/dates.py

```diff
@@ -444,7 +444,9 @@ def date2num(d):
     if not iterable:
         d = [d]
 
-    d = np.asarray(d)
+    masked = np.any(np.ma.getmask(d))
+    d = np.ma.asarray(d) if masked else np.asarray(d)
```

Comment:
```suggestion
    d = np.asanyarray(d)
```
?

Reply:
Very possibly so. I have, clearly, limited knowledge of masked arrays...